test/integration/com/pensio/api/PensioMerchantAPITest.java

```diff
@@ -58,6 +58,21 @@ public void createPaymentRequest() throws Throwable
 		assertNotNull(result.getUrl());
 	}
 
+	@Test
+	public void createPaymentRequestWithAgreement() throws Throwable
+	{
+        AgreementConfig agreementConfig = new AgreementConfig();
+		agreementConfig.setAgreementType(AgreementType.unscheduled);
+        agreementConfig.setAgreementUnscheduledType(AgreementUnscheduledType.incremental);
+
+		PaymentRequestResponse result = api.createPaymentRequest(
+				new PaymentRequest(getOrderId(), "AltaPay Test Terminal", Amount.get(7.77, Currency.EUR))
+						.setAgreementConfig(agreementConfig)
+		);		
+
+		assertNotNull(result.getUrl());
```

Comment:
asserting that `result.getUrl()` is not null doesn't speak much about what we're actually testing here

maybe instead we could call `getAPIResponse()` method and make sure that it returns a valid response without any exceptions?

Reply:
`result.getUrl()` will return a URL which is the most important response required and is returned in case of success only. So, its safe to use it as a way of detecting success.